crates/config/src/errors.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright 2023 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+pub type Result<T> = std::result::Result<T, ConfigError>;
+
+#[derive(Debug)]
+pub enum ConfigError {
+    CouldNotLoadConfig(std::io::Error),
+    CouldNotParseConfig(toml::de::Error),
+    CouldNotSaveConfig,
+}
```

Comment:
I'm curious about this. Are there any recommendations in Rust for naming errors? I found that different crates follow different approaches. 

I also found that the `runtimes` crate uses `Cannot` instead of `CouldNot`. I believe that's more correct and I would update these error names too:

```suggestion
pub enum ConfigError {
    CannotLoadConfig(std::io::Error),
    CannotParseConfig(toml::de::Error),
    CannotSaveConfig,
}
```

Reply:
Besides what is documented at https://rust-lang.github.io/api-guidelines/naming.html I'm not aware of any. I guess it depends a lot on the project too, so we are fine as long as we are consistent with our own naming etc, as you have correctly observed and corrected in this PR :)